### PR TITLE
Feature/components

### PR DESCRIPTION
**Pull request by pacyL2K19**
# The aim of this PR 
- Creation of the global state 
- Implementation of actions
- Implementation of reducers
- Consumption of the global state in the child components

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -18,7 +18,8 @@
     "react/jsx-filename-extension": ["warn", { "extensions": [".js", ".jsx"] }],
     "react/react-in-jsx-scope": "off",
     "import/no-unresolved": "off",
-    "no-shadow": "off"
+    "no-shadow": "off",
+    "jsx-a11y/label-has-associated-control": 0
```
Comment (bellom) [inline .eslintrc.json:22]: Opps, we have some little changes in this file, check out the [microverse-linter-config](https://github.com/microverseinc/linters-config/blob/master/html-css-js/.eslintrc.json)
Kindly make use of the files provided by Microverse without editing the file.

File: src/containers/BooksForm.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      <select id="cars" name="cars">
+        <option value="action">{action}</option>
+        <option value="biography">{biography}</option>
+        <option value="history">{history}</option>
+        <option value="horror">{horror}</option>
+        <option value="kids">{kids}</option>
+        <option value="learning">{learning}</option>
+        <option value="sci-fi">{scifi}</option>
+      </select>
+      <button type="submit">Save</button>
```
Comment (bellom) [inline src/containers/BooksForm.js]: Let's create a constant with Books categories then we can map through it.
image


Example:
```
<select>
        {book.map((x, i) => (.....}
 </select>
```

File: src/index.js [hunk 3]
```diff
@@ -1,16 +1,34 @@
...
+    {
+      title: 'Learn React',
+      category: 'Learning',
+      id: Math.floor(Math.random() * 1000),
+    },
+    {
+      title: 'Learn Redux',
+      category: 'Learning',
+      id: Math.floor(Math.random() * 1000),
+    },
```
Comment (bellom) [inline src/index.js]: ### [Optional]
Kindly be aware that you can use uuidv4 to automatically generate ids. read more 👉 [npm uuid](https://www.npmjs.com/package/uuidv4)

Example:
```
import uuid from 'uuidv4'
id: uuid(): {title: 'Learn React', category: 'Learning'}
```

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**1. Kindly check my comments under this review;**

 [ ]  [REQUIRED] `index.js` reducer [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/ReducxReducers/CombinesAppReducers.png) from [pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896827/). it is placed in `src/reducers/index.js`.

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best**

**Review by NtwaliHeritier — APPROVED** [review]
### **STATUS: APPROVED👍️👍️**

Hello team,
Great work making the required changes👏️👏️

You can merge👍️👍️

Happy coding💻️💻️

**Pull request merged**